src/qt/sendcoinsdialog.cpp

```diff
@@ -630,6 +630,9 @@ void SendCoinsDialog::useAvailableBalance(SendCoinsEntry* entry)
         coin_control = *CoinControlDialog::coinControl();
     }
 
+    // Include watch-only for wallets without private key
+    coin_control.fAllowWatchOnly = model->privateKeysDisabled();
```

Comment:
eef2b6350684f2675989b48a2d7f79c0d8a7be6e

Does this have to be here?  Couldn't this be checked in `CWallet::AvailableCoins`?

Reply:
I don't think that's a good idea. The RPC sets this based on `includeWatching`, so then we'd have to make it an `Optional`.

Reply:
Good point.